.github/workflows/workflow.yml

```diff
@@ -21,9 +21,67 @@ jobs:
     - name: Run Tox Check
       run: tox -e check
 
-  build:
+  data-cache:
     needs: check
     runs-on: ubuntu-latest
+    outputs:
+      cache_hash: ${{ steps.compute-cache-hash.outputs.cache_hash }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e ".[dev]"
+      - name: Restore Cache
+        uses: actions/cache@v4
+        with:
+          key: cached-data-${{ hashFiles('**/registry.py', '**/*workflow.yml') }}
+          restore-keys: |
+            cached-data-
+          path: .github_cache/ASPIRE-data
+          enableCrossOsArchive: true
+      - name: Create Cache Directory
+        run: |
+          # Create cache and config directories
+          mkdir -p .github_cache/ASPIRE-data
+          chmod -R 777 .github_cache/ASPIRE-data
+          echo "cache:" > config.yaml
+          echo "    cache_dir: .github_cache/ASPIRE-data" >> config.yaml
+          echo "Logging config.yaml for verification:"
+          cat config.yaml
+      - name: Download Cache Files
+        run: |
+          export ASPIREDIR=.
+          python -c "
+          from aspire.downloader import emdb_2660, simulated_channelspin
+          emdb_2660()
+          simulated_channelspin()
+          "
+      - name: Compute Cache Directory Hash
+        id: compute-cache-hash
+        run: |
+          echo "Computing hash for .github_cache/ASPIRE-data..."
+          # Compute a hash for each file, sort the results, and then hash that list.
+          cache_hash=$(find .github_cache/ASPIRE-data -type f -exec sha256sum {} \; | sort | sha256sum | awk '{print $1}')
```

Comment:
Can we not use the GH hashfiles at this time?

If we need to keep... I'm fairly confident you can get rid of the `awk` by using a `--quiet` to just print the sum.  Also `md5` is totally adequate here and might be faster.  I would not have mentioned except that you may end up changingt the line anyway.

Reply:
Note, that I suspect you only need to hash the recursive dir listing, not the file contents, for our purposes.  (I am assuming we can trust pooch to catch any known files that don't match it's registry, so we just need to know if new files were written).

Reply:
I don't think we can use GH hashFiles here. As far as I can tell, hashFiles is evaluated when the workflow is parsed and changes made to files during runtime won't be picked up. 

Switched to `md5` with `--quiet` on just the file listing. Thanks for the suggestions.

Reply:
Cleaned this up to: `ls -1 .github_cache/ASPIRE-data/** | md5sum`

Reply:
> I don't think we can use GH hashFiles here. As far as I can tell, hashFiles is evaluated when the workflow is parsed and changes made to files during runtime won't be picked up.
> 
> Switched to `md5` with `--quiet` on just the file listing. Thanks for the suggestions.

discussed. Not sure about the hashfiles, I got the impression it was designed for the cases similar to we're considering. You are correct it could not be applied prior to checking out (or downloading) the files. Anyway, looks you went the simpler approach I suggested of just using the dir listing.   This is much cleaner anyway. Nice.